unittests/snapshot_tests.cpp

```diff
@@ -384,4 +385,60 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(test_compatible_versions, SNAPSHOT_SUITE, snapshot
    }
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(test_pending_schedule_snapshot, SNAPSHOT_SUITE, snapshot_suites)
+{
+   tester chain(setup_policy::preactivate_feature_and_new_bios);
+   auto block = chain.produce_block();
+   BOOST_REQUIRE_EQUAL(block->block_num(), 3); // ensure that test setup stays consistent with original snapshot setup
+   chain.create_account(N(snapshot));
+   block = chain.produce_block();
+   BOOST_REQUIRE_EQUAL(block->block_num(), 4);
+
+   const auto& gpo = chain.control->get_global_properties();
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.version, 0);
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.producers.size(), 0);
+
+   auto res = chain.set_producers_legacy( {N(snapshot)} );
+   block = chain.produce_block();
+   BOOST_REQUIRE_EQUAL(block->block_num(), 5);
+   chain.control->abort_block();
+   ///< End deterministic code to generate blockchain for comparison
+   auto base_integrity_value = chain.control->calculate_integrity_hash();
+
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.version, 1);
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.producers.size(), 1);
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.producers[0].producer_name.to_string(), "snapshot");
+
+   static_assert(chain_snapshot_header::minimum_compatible_version <= 2, "version 2 unit test is no longer needed.  Please clean up data files");
+   auto v2 = SNAPSHOT_SUITE::template load_from_file<snapshots::snap_v2_prod_sched>();
+   snapshotted_tester v2_tester(chain.get_config(), SNAPSHOT_SUITE::get_reader(v2), 0);
+   auto v2_integrity_value = v2_tester.control->calculate_integrity_hash();
+   if (std::is_same_v<SNAPSHOT_SUITE, variant_snapshot_suite> && v2_integrity_value.str() != base_integrity_value.str()) {
```

Comment:
This conditional means that we are not calling `BOOST_REQUIRE_EQUAL(v2_integrity_value.str(), base_integrity_value.str());` on the binary form of the snapshots below and therefore not actually validating that the binary snapshot is loadable with the same integrity value.